python/src/trezorlib/client.py

```diff
@@ -147,8 +151,9 @@ def _raw_read(self):
         )
         return msg
 
-    def _callback_pin(self, msg):
+    def _callback_pin(self, msg: messages.PinMatrixRequest) -> "MessageType":
         try:
+            assert msg.type is not None
```

Comment:
this assert is wrong. instead, the type should be marked `Optional` in the protocol definition

Reply:
Done in 7d76466

Reply:
I probably made a mistake here, I misunderstood it to make it `required` in protobuf (which made sense to me, as it is the only field in this message). Just to be sure, you meant to change the `get_pin()` to have the argument as `Optional`?

Reply:
correct. I said "protocol" because `TrezorUI` is a protocol type.

I believe the type is not always sent by Trezor. We _could_ modify the protobuf definition and add a zero default, but that's out of scope here.

Reply:
Done in 0c94cfcb5